### PR TITLE
Averaged test

### PR DESCRIPTION
**Pull request by Antoine-marchais**
refactor test pipeline to average result over the whole spectrogram

### CODE REVIEW

File: utils.py [hunk 1]
```diff
@@ -57,6 +72,16 @@ def save_model(model, epoch):
     model.save_weights(os.path.join(
         config.SAVED_MODELS_PATH, f"{model.name}_{epoch:06d}{dev_mode_string}.h5"))
 
+def load_model(file_path, batch_size=1):
+    type = os.path.split(file_path)[-1].split("_")[0]
+    if type == "BasicCnn":
+        model = BasicCNN()
+    if type == "PianocktailGru":
+        model = PianocktailGRU()
```
Comment (quentinverlhac) [inline utils.py]: I think we could use the enum that is defined in `config.py` here. Maybe we could move this enum in a separate file in model folders.
Let's do it in another PR though.

File: train.py [hunk 2]
```diff
@@ -1,30 +1,100 @@
...
+    with tf.GradientTape() as tape:
+        predictions_, loss = forward_pass(inputs, labels, model)
+    gradients = tape.gradient(loss, model.trainable_variables)
+    optimizer.apply_gradients(zip(gradients, model.trainable_variables))
+
+    # compute metrics
+    train_loss.update_state(loss)
+    train_accuracy.update_state(labels, predictions_)
+
+    return predictions_, labels
```
Comment (quentinverlhac) [inline train.py]: By the way, what does the `_` at the end of `prediction_` mean here ?

Comment (owain-Biddulph) [reply train.py]: I added it because the function used to be defined in the train function where a predictions variable is used. I forgot to change it when the function was separated.
